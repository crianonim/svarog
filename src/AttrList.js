import React from "react";

const AttrList = ({element,attrs}) => {
  return (
    <div className="flex-row flex-wrap attr-wrapper block">
      <span>{element}</span>
      
      {Object.entries(attrs).map(([key, value], i) => {
        return (
          <span key={key} className="attr-pair">
            <span className="key-name">{key}</span>
            <span className="">{value}</span>
          </span>

        );
      })}
    </div>
  );
};


export default AttrList;
