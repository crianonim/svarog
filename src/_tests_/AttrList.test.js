import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AttrList from "../components/AttrList";

it("matches a snapshot",()=>{
    const wrapper=shallow(<AttrList element="circle" attrs={{"x":10,"y":20,"r":30}}/>) 
    expect(toJson(wrapper)).toMatchSnapshot();
});

it("has a shape name",()=>{
    const wrapper=shallow(<AttrList element="circle" attrs={{"x":10,"y":20,"r":30}}/>) 
    expect(wrapper.find('div > span:first-child').text()).toBe("circle")
})

it("has 3 attributes",()=>{
    const wrapper=shallow(<AttrList element="circle" attrs={{"x":10,"y":20,"r":30}}/>) 
    const attrs=wrapper.find('div > span').slice(1)
    expect(attrs.length).toBe(3)
})