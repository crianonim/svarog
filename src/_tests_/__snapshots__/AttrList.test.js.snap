// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<div
  className="flex-row flex-wrap attr-wrapper block"
>
  <span>
    circle
  </span>
  <span
    className="attr-pair"
    key="x"
  >
    <span
      className="key-name"
    >
      x
    </span>
    <span
      className=""
    >
      10
    </span>
  </span>
  <span
    className="attr-pair"
    key="y"
  >
    <span
      className="key-name"
    >
      y
    </span>
    <span
      className=""
    >
      20
    </span>
  </span>
  <span
    className="attr-pair"
    key="r"
  >
    <span
      className="key-name"
    >
      r
    </span>
    <span
      className=""
    >
      30
    </span>
  </span>
</div>
`;
